map-widget/qmlmapwidgethelper.cpp

```diff
@@ -4,20 +4,20 @@
 #include <QDebug>
 #include <QVector>
 
-#include "qmlmapwidgethelper.h"
 #include "core/divefilter.h"
 #include "core/divelist.h"
 #include "core/divelog.h"
 #include "core/divesite.h"
 #include "core/qthelper.h"
 #include "core/range.h"
-#include "qt-models/maplocationmodel.h"
+#include "qmlmapwidgethelper.h"
 #include "qt-models/divelocationmodel.h"
+#include "qt-models/maplocationmodel.h"
 #ifndef SUBSURFACE_MOBILE
 #include "desktop-widgets/mapwidget.h"
 #endif
 
-#define SMALL_CIRCLE_RADIUS_PX            26.0
+#define SMALL_CIRCLE_RADIUS_PX 26.0
```

Comment:
I don't get the point of these whitespace changes.

If you want to touch this line, at least transform it into a proper constant, such as
```
static constexpr double small_circle_radius_px = 20.0;
```

Reply:
> I don't get the point of these whitespace changes.

The point is, that there is no point, I just made some quick changes and didn't notice I had the "prettier" plugin enabled, which autosaves files and changes whitespaces/indentation etc.

So that's all. And yes you are right I should have changed it back, but I didn't

Reply:
> I don't get the point of these whitespace changes.
> 
> If you want to touch this line, at least transform it into a proper constant, such as
> 
> ```
> static constexpr double small_circle_radius_px = 20.0;
> ```

You know, I liked Egbert's change better. He at least didn't change the value :)

But joking aside - I had pointed that out in my first review and we established it was unintentional. All good.

Of course, I am the first to admit that I can be really sensitive on whitespace - and I'm super annoyed that new integrated IDEs all want to change the whitespace in my files... so one of the reasons I'm having posted a quick snapshot of what I have is that it's a complete mess, whitespace wise, because Qt Creator somehow decided to change things even though I have our settings loaded... grrr...

Reply:
Since we already have a discussion about whitespace going here this seems to be a good place to chip in. 😉

And I think Egbert has got a point here - probably somewhat unintentional, but a point nonetheless:
If we want to enforce whitespace standards, can we make them something that is structured enough that it can be tested (or even enforced) by a linter like `prettier`?

In the past I've always found that formatting rules that avoid dependencies on context (in the context of whitespace: vertical alignment) result in less 'noise' coming from code changes, and more compact pull requests - and this is typically what can be achieved relatively easy with linters.

Reply:
We have a Qt Creator config in the coding standard document, iirc.
I'm happy to consider changes, but those need to be easily implementable for people using different tools on different platforms. 
We have developers on Linux and Mac (I doubt anyone is on Windows), and people use a wife array of tools. uemacs, vim, Qt Creator, VSC, etc. 
So yes, by all means, let's make things consistent, but don't break things for the people who have worked on this project for more than a dozen years...

Reply:
> We have a Qt Creator config in the coding standard document, iirc.

Not sure if it makes sense to ask everybody contributing to Subsurface to use Qt Creator?

> I'm happy to consider changes, but those need to be easily implementable for people using different tools on different platforms. We have developers on Linux and Mac (I doubt anyone is on Windows), and people use a wife array of tools. uemacs, vim, Qt Creator, VSC, etc.

In general I think it will make sense to go for a standalone tool like Prettier or `clang-tidy`, that can be run by developers independently of the IDE they are using, and can also be run in a GitHub Action, to automate pull request feedback about coding style.

> So yes, by all means, let's make things consistent, but don't break things for the people who have worked on this project for more than a dozen years...

I am a bit on the fence on this one - while I see the value of retaining _'we've always done it this way'_ workflows, I also see appeal in moving to a coding style that can be automated with tooling, and minimises the amount of effort and friction for our workflows around pull requests. And I suspect that there won't be an approach that fully satisfies both of these.

Reply:
And in all honesty, the people who started this project (those still active) are not C++ programmers and the second generation of people engaged has already taken the project in a direction none of us are very familiar with. 

Half the code I can barely make sense of, most of the time. Which is a big part of why I have written less and less code in the last few years and mostly focused on the backend and infrastructure.

And after battling the very disruptive changes that Qt6 brings to QML projects, I'm not sure how much more I'll be able to constructively contribute. It seems that the fundamental way in which we integrate C++ code in QML has all been replaced by mechanisms that are designed around the idea that you have a QML app that gets a few additional features with dedicated Qt code. Not a C/C++ app that has a QML UI.

Reply:
> And in all honesty, the people who started this project (those still active) are not C++ programmers and the second generation of people engaged has already taken the project in a direction none of us are very familiar with.

I am not a C++ programmer either - but then I've always seen myself as a coder in <insert language that can get me what I want in the most effective way>, and (mostly) enjoyed the learning that came from it.

> And after battling the very disruptive changes that Qt6 brings to QML projects, I'm not sure how much more I'll be able to constructively contribute. It seems that the fundamental way in which we integrate C++ code in QML has all been replaced by mechanisms that are designed around the idea that you have a QML app that gets a few additional features with dedicated Qt code. Not a C/C++ app that has a QML UI.

I think this is more a 'sign of the times' than a direction that is particular to Qt / QML. html / CSS / JavaScript has become the 'lingua franca' that drives pretty much all multi - platform UI development, and QML is just the glue logic that goes in between Qt / C++ and the UI framework. And while using this isn't always a pleasurable experience, I suspect that this is something that is inherent to any UI design of sufficient complexity, and even more so if it strives to be multi-platform.

But to get back on topic, I think a sensible first step in the direction of automating our style guide will be to set up a GitHub action that checks changed lines in pull requests against a guide, and returns the result as a recommendation (i.e. doesn't fail) - we can then start with a template set of rules, and refine it to where we want it.
Thoughts?

Reply:
> Half the code I can barely make sense of, most of the time.

If something is unclear, you are always welcome to ask.

Concerning white space, different people have different styles and I think that is OK. The current rough guidelines seem adequate to me.

When changing a source file, just use the overall style in there.

I have no problem with the odd whitespace fix in a commit (if it is consistent with the guidelines). However, when half of it are whitespace fixes, that makes reviewing just unnecessarily painful. Don't do it, please.

> It seems that the fundamental way in which we integrate C++ code in QML has all been replaced by mechanisms that are designed around the idea that you have a QML app that gets a few additional features with dedicated Qt code. Not a C/C++ app that has a QML UI.

If one can't have a thin display layer of QML on top of a C++ app, that's the point where I would start thinking about alternatives. Though I doubt it.

Reply:
> If one can't have a thin display layer of QML on top of a C++ app, that's the point where I would start thinking about alternatives. Though I doubt it.

I think you are right. There is more separation of GUI and backend code. Also everything is becoming more web based/web language oriented. QML is in fact just a QT Javascript front end language. A lot of apps nowadays are written in a web framework and brought to the desktop as an electron app.  

Even Vscode itself is an Electron app, which in fact is nothing more than a web app packaged together with a dedicated stripped browser.

Back on topic, the Usermanual seems no problem to get it working, searching etc. is working, the only problem I have at the moment is that we need to overload the class QWebEngineView function acceptNavigationRequest  because setLinkDelegationPolicy function isn't supported anymore. 

Concerning the printing a have been testing some code to use the  runJavaScript() function because the findAllElements() isn't supported anymore, I have it somewhat working, but it isn't pretty. Maybe it's better and more readable to parse the HTML as HTML text and do some styling and insert the profile image instead of trying to use the Webview to do it.